hera_cal/tests/test_nucal.py

```diff
@@ -561,3 +558,261 @@ def test_fit_nucal_foreground_model():
         assert u_model_comps[k].shape[0] == ntimes
         assert model_comps[k].shape[1:] == (spectral_filters.shape[-1], spatial_filters[k].shape[-1])
         assert u_model_comps[k].shape[1:] == (spatial_filters[k].shape[-1],)
+
+class TestGradientDescent:
+    def setup_method(self):
+        self.freqs = np.linspace(50e6, 250e6, 200)
+        self.antpos = linear_array(10, sep=2)
```

Comment:
I'm wondering if you're missing any important edge cases by only testing on a 1D array. Can you also do a small hex array or is that too slow?

Reply:
I think it's probably okay to just leave this as a 1-D array for the purposes of testing, considering these tests will be run whenever anything in `hera_cal` gets updated. Since we already wrote the paper on the full array, I don't feel so bad leaving out the 2D case. I'm open to considering a 2D array if you feel that it should be added.

Reply:
As a compromise, let's do two linear arms